client/llb/exec.go

```diff
@@ -142,6 +143,11 @@ func (e *ExecOp) Marshal(c *Constraints) (digest.Digest, []byte, *pb.OpMetadata,
 			e.meta.Env = e.meta.Env.AddOrReplace("SSH_AUTH_SOCK", e.ssh[0].Target)
 		}
 	}
+	if err := c.Caps.Supports(pb.CapExecMetaSetsDefaultPath); err != nil {
+		e.meta.Env.SetDefault("PATH", system.DefaultPathEnv)
```

Comment:
This shouldn't work because `meta.Env` is not a pointer.

Reply:
Urk, I think I need to redouble my efforts to construct a test case then!